[docs] Contrast built-in and external providers (#461)
wasmCloud/wasmCloud.com: docs/concepts/providers.mdx

## Overview

> **Providers** are executable [host](/docs/concepts/hosts) plugins dedicated to longer-lived processes fulfilling non-functional requirements (sometimes called **capabilities**). Providers are typically responsible for functionality that is not considered part of the core business logic of an application, such as...

Reviewer: ```suggestion
**Providers** are executable binaries that function as plugins for the [wasmCloud host](/docs/concepts/hosts).

Providers run as long-lived processes providing business-logic agnostic functionality sometimes referred to as **capabilities** (hence the term "capability provider"), such as:
```

> - Sending notifications
> - Fetching secret values
> - Accessing databases
> - Serving content over HTTP

Reviewer: ```suggestion
- Accessing databases
- Sending notifications
- Fetching secrets
- Serving content over HTTP
```

> For example, when writing a web service that exposes a [RESTful API](https://en.wikipedia.org/wiki/REST), the HTTP server and the database are not parts of your business logic&mdash;your primary concern is structuring your API so that users can easily get the data they need. If you were to write this web service in a component, you would depend on providers to fulfill HTTP and database functionality in order to execute your business logic. In wasmCloud, providers are self-contained processes that implement a well-defined API.

Reviewer: ```suggestion
**Components that contain business logic use the capabilities offered by providers.**

For example, when writing a web service that exposes a [RESTful API](https://en.wikipedia.org/wiki/REST), the HTTP server and the database are not parts of your business logic&mdash;your primary concern is structuring your API so that users can easily get the data they need. If you were to write this web service in a component, you would depend on providers for non-business-logic capabilities:

 - HTTP server (to serve HTTP requests)
 - Database access (to save changes made via the interface)
 
In wasmCloud, providers implement APIs (WIT interfaces) that enable functionality like the case described above.
```

You can find a complete list of first-party providers on the [wasmCloud GitHub Packages page](https://github.com/orgs/wasmCloud/packages?repo_name=wasmCloud). First-party providers are named to specify the [interface](/docs/concepts/interfaces) being used and the specific functionality implemented with the interface (e.g. "keyvalue" for the WASI interface and "Redis" for the specific functionality).

> First-party providers may be **external** or **built-in**. External providers run on the wasmCloud [lattice](/docs/concepts/lattice/) like components&mdash;they are "external" in the sense that they are external to any given wasmCloud host. All third-party providers are external, and generally, when we talk about providers, we are talking about external providers. There is, however, a small set of **built-in** providers that represent wasmCloud's trusted compute base and are built directly into the wasmCloud host. 

Reviewer: ```suggestion
First-party providers may be **external** or **built-in**. External providers run on the wasmCloud [lattice](/docs/concepts/lattice/) like components&mdash;they are "external" in the sense that they are external to any given wasmCloud host. 

All third-party providers are external, and generally, when we talk about providers, we are talking about external providers. There is, however, a small set of **built-in** providers that represent wasmCloud's trusted compute base and are built directly into the wasmCloud host. 
```

### External providers

Since external providers are standalone entities on the lattice, they may be scaled and updated independently of any given wasmCloud host. 

External first-party providers include:

> - [http-client](https://github.com/wasmCloud/wasmCloud/tree/main/crates/provider-http-client)
> - [http-server](https://github.com/wasmCloud/wasmCloud/tree/main/crates/provider-http-server)

Reviewer: ```suggestion
- [http-client](https://github.com/wasmCloud/wasmCloud/tree/main/crates/provider-http-client) - provides *outgoing* HTTP requests to the internet
- [http-server](https://github.com/wasmCloud/wasmCloud/tree/main/crates/provider-http-server) - allows components to handle incoming HTTP requests by listening on a local port
```

In [Wadm manifests](/docs/ecosystem/wadm/), you can refer to the [wasmCloud GitHub Packages registry](https://github.com/orgs/wasmCloud/packages?repo_name=wasmCloud) for OCI images of providers&mdash;for example: `ghcr.io/wasmcloud/keyvalue-redis:0.23.0`

### Built-in providers

> Because built-in providers are part of the wasmCloud host itself, they do not communicate over NATS and run in the same process as the host. Updating a built-in provider requires updating the host. Built-in providers include:

Reviewer: ```suggestion
Built-in providers are part of the wasmCloud host itself, they do not communicate over NATS and run in the same process as the host. Updating a built-in provider requires updating the host. 

Built-in providers include:
```

* Logging (using the [`wasi-logging`](https://github.com/WebAssembly/wasi-logging) interface)
* Random (using the [`wasi-random`](https://github.com/WebAssembly/wasi-random) interface)
* Clocks (using the [`wasi-clocks`](https://github.com/WebAssembly/wasi-clocks) interface)

> Built-in providers form wasmCloud's **trusted compute base**&mdash;carefully vetted code providing core functionalities. You can learn more about using built-in providers on the [Linking at runtime](/docs/concepts/linking-components/linking-at-runtime#built-in-providers) page.

Reviewer: ```suggestion
Built-in providers form wasmCloud's **trusted compute base**&mdash;carefully vetted code providing core functionalities. 

You can learn more about using built-in providers on the [Linking at runtime](/docs/concepts/linking-components/linking-at-runtime#built-in-providers) page.
```
